testing/citest/tests/kube_v2_helm_test.py

```diff
@@ -0,0 +1,333 @@
+    return agent
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    """Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    """
+    super(KubeV2HelmTestScenario, cls).initArgumentParser(
+        parser, defaults=defaults)
+
+    defaults = defaults or {}
+    parser.add_argument(
+      '--test_namespace', default='default',
+      help='The namespace to manage within the tests.')
+
+    parser.add_argument(
+        '--test_gcs_artifact_account',
+        default='gcs-artifact-account',
+        help='Spinnaker GCS artifact account name to use for test operations'
+             ' against artifacts stored in GCS.')
+
+    parser.add_argument(
+        '--test_gcs_bucket',
+        help='GCS bucket to upload & read manifests from.')
+
+    parser.add_argument(
+        '--test_gcs_credentials_path', default=None,
+        help='GCS bucket to upload & read manifests from.')
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(KubeV2HelmTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']
+
+    # Take just the first if there are multiple
+    # because some uses below assume just one.
+    self.TEST_NAMESPACE = bindings['TEST_NAMESPACE'].split(',')[0]
+
+    self.ARTIFACT_ACCOUNT = bindings['TEST_GCS_ARTIFACT_ACCOUNT']
+    self.BUCKET = bindings['TEST_GCS_BUCKET']
+
+    self.mf = sk.KubernetesManifestFactory(self)
+    self.mp = sk.KubernetesManifestPredicateFactory()
+    self.gcs = sk.GcsFileUploadAgent(bindings['TEST_GCS_CREDENTIALS_PATH'])
+    self.ps = sk.PipelineSupport(self)
+
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_create_app_operation(
+            bindings=self.bindings, application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_KUBERNETES_V2_ACCOUNT']),
+        contract=contract)
+
+  def delete_app(self):
+    """Creates OperationContract that deletes a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_KUBERNETES_V2_ACCOUNT']),
+        contract=contract)
+
+  def __create_helm_chart(self):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    command = 'helm package {}/kube_v2_data/helm-test'.format(dir_path)
+    subprocess.Popen(command, stderr=sys.stderr, shell=True).wait()
```

Comment:
There should probably be a helper for running commands like this so you don't have to manage streams and waiting and such. We have a nice set of helper functions in `buildtool`, don't know if we want to include that lib in `testing/citest` though.

Reply:
What would the helper do beyond calling subprocess.Popen(..).wait()? It feels pretty encapsulated already

Reply:
Cleans up the interface and provides logging, etc. [What I was referring to](https://github.com/spinnaker/spinnaker/blob/ae4006af07c08fc9259dd7c071bc7a39571cc6f9/dev/buildtool/subprocess_support.py#L130). May be overkill for one case but if this is more systematic then may be worth it.